### ISSUE TITLE
Installation instructions/UX utterly frustrating; "trusted" install fails on macOS

### ISSUE THREAD

**Issue by tgrushka** [tcdi/plrust#415]
TLDR: I found the following works for me on macOS Sequoia arm64 to install the untrusted version locally for development (cannot get trusted to compile at all; running build script fails when a .json file cannot be found).

```
rustc -V # => rustc 1.83.0 (90b35a623 2024-11-26)
cargo install --version 0.11.0 cargo-pgrx
cargo pgrx init --pg16 /opt/homebrew/bin/pg_config
git clone https://github.com/tcdi/plrust.git ~/src/plrust
cd ~/src/plrust
rm rust-toolchain.toml
cd plrust
cargo pgrx install --release -c /opt/homebrew/bin/pg_config
```

After a few hours going back and forth between the README and the two different doc pages, I nearly gave up. These docs were obviously *not* written for someone who is not a contributor/maintainer to this project, and surprisingly convoluted for a Rust project. I recall installation about 8 months ago to be far simpler, at least on Linux... what in Crustacia happened?

1. Requiring a minimum Rust version is normal. However, requiring a specific system default Rust version is just really bad. Rust is backwards-compatible. There is no reason for requiring switching the system default, especially because pgrx now depends on a later version of rust and won't even compile with 1.72.0 anymore (it now requires a feature that was experimental in that version).
2. The `plrust/plrust/build` script "clobbers my configuration" (even though a comment inside acknowledges that it shouldn't!), even following your instructions to a tee. Once it runs and fails, the rust standard library and everything else is gone. Only a `rustup self uninstall && rm -rf ~/.pgrx` followed by clean install of rustup fixes it. Tried several times with same result. I've never seen a Rust crate build process that kills Rust itself! `cargo pgrx install ...` really should be sufficient for either trusted or untrusted version. It should just be a feature flag issue with the proper crates being imported in Cargo.toml.
3. It seems the only real difference between untrusted and trusted is an extra crate (`postgrestd` -- which itself won't even compile for me with rustup config specified in docs) that trusted uses to prevent system calls. This could be summarized in a sentence on the installation instructions page.
4. Link to installation instructions skips whole prerequisites page. It's totally unclear what the prerequisites are for, but they seem only really necessary for trusted installation.

Is there any way to simplify the README and docs for those of us who are not plrust hackers (most of us)?

To reproduce failure, just try installing trusted version on an Apple Silicon Mac following README and docs, with a clean install of rustup/rustc.

**Comment by pinda** [comment]
What did you do to get it to work?
